code_curation2_submissions2/Julian_clustering_pipeline.R

```diff
@@ -0,0 +1,246 @@
+        ## adasyn correction 
+        warning('Artificial cases have to be generated as group sizes between smallest and biggest group differ by more than 50%')
+        optim_df$adasyn[param] <- 1
+        # neuropsy data
+        num_k_min <- min(table(data_clustering$cluster)) - 1  # k minimum is one less than cluster size of minimum cluster where cases are generated for
+        num_k <- ifelse(num_k_min >= 5, 5,num_k_min) # if minimum number of k > 5 --> k=5 (recommended by default); if it is lower, than minium k has to be chosen otherwise algorithm crashes
+        data_clustering_adasyn <- AdasynClassif(cluster ~ ., data_clustering[,c(neuropsy_ind, clinical_ind, 142)], k=num_k, beta=1)
+        data_clustering_class <- data_clustering_adasyn[, c(84:101)]     # ATTENTION HARD CODE!
+        optim_df$cluster_size_adasyn[param] <- list(table(data_clustering_class$cluster))
+      } else {
+        optim_df$adasyn[param] <- 0
+        data_clustering_class <- data_clustering[, c(clinical_ind, 142)]
+        optim_df$cluster_size_adasyn[param] <- list(table(data_clustering_class$cluster))
+      }
+      
+      # svm classification
+      svmfit <- svm(cluster ~ ., data = data_clustering_class, kernel = "linear", cost = 1, scale = T)
+      conf_mat <- confusionMatrix(data_clustering_class$cluster, predict(svmfit))
+      optim_df$BAC[param] <- conf_mat$overall[1]
+      optim_df$BAC_sig[param] <- ifelse(conf_mat$overall[6] < 0.05, 1,0)    # 1= significant p value
+    
+    
+    } else if(min(table(data_clustering$cluster)) < (round(nrow(data_clustering)*0.05))){
+      message('Mininum cluster size to calculate classification not reached.')
+      optim_df$adasyn[param] <- 0
+      optim_df$BAC[param] <- NA
+      optim_df$BAC_sig[param] <- NA
+      optim_df$cluster_size_adasyn[param] <- list(table(data_clustering$cluster))
+    }
+  }
+  
+  optim_df_total <- rbind(optim_df_total, optim_df)
+}
+
+
+#######################
+## plotting of results
+#######################
+
+optim_df_total <- as.data.frame(optim_df_total) 
+optim_df_total$clust.alg <- as.factor(paste(optim_df_total$num_cluster, optim_df_total$clust_alg, sep = "."))
+optim_df_total <- summaryBy(stability_subset + stability_noise ~ clust.alg + permut, optim_df_total, FUN=mean)
+optim_df_total$permut <- as.factor(optim_df_total$permut)
+
+## plot stability values of resampling method 'subset' across all permutations
+ggplot(optim_df_total, aes(x= permut, y=stability_subset.mean, group=clust.alg, colour=clust.alg)) + 
+  geom_line() +
+  geom_point()
+
+## plot stability values of resampling method 'noise' across all permutations
+ggplot(optim_df_total, aes(x= permut, y=stability_noise.mean, group=clust.alg, colour=clust.alg)) + 
+  geom_line() +
+  geom_point()
+
+

```

Comment:
Well documented!
Except no clear definition of User defined variables or constants.